File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.java
public class Codec {
    // the key here is to use a delimeter which is on an ascii value. 
    // Example: π 
    

    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
        StringBuilder encoded_string = new StringBuilder();
        for(String str : strs) {
            encoded_string.append(str).append("π"); 
        }
        System.out.println(encoded_string.toString()); 
        return encoded_string.toString(); 
    }

    // Decodes a single string to a list of strings.
    public List<String> decode(String s) {
        List<String> strs = new ArrayList<>();
        StringBuilder sb = new StringBuilder(); 
        for(int i = 0; i < s.length(); i++) {
            char c = s.charAt(i); 
            if(c != 'π') {
                sb.append(c); 
            }else{
                // end of the string
                strs.add(sb.toString()); 
                sb.setLength(0);
            }
        }
        return strs; 
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(strs));

File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        res=""
        for s in strs:
            res += str(len(s)) + "#" + s
        return res
            
        

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """
        list, i = [], 0
        while (i < len(s)):
            j = i
            while(s[j] != "#"):
                j +=1
            length_of_word = int(s[i:j])
            list.append(s[j + 1 : j + 1 + length_of_word])
            i = j + 1 + length_of_word
        return list
            
            
        
        


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))